File: Language/C#/Book/tmp/LINQ实战/LinqInAction/Chapter06to08.CSharp/Common/CodeSamples/EditingForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using LinqInAction.Chapter06to08.Common.SampleClasses.Ch8;

namespace LinqInAction.Chapter06to08.Common.CodeSamples
{
  public partial class EditingForm : Form
  {
    public EditingForm()
    {
      InitializeComponent();
    }

    private void EditingForm_Load(object sender, EventArgs e)
    {
      Ch8DataContext context = new Ch8DataContext();
      var query = from p in context.Publishers
                  select p;
      this.publisherBindingSource.DataSource = query.ToList();

    }
  }
}

File: Language/C#/Book/tmp/LINQ实战/Chapter06to08.VB/Common/CodeSamples/EditingForm.vb
Imports LinqInAction.Chapter06to08.Common.SampleClasses.Ch8
Namespace CodeSamples
    Public Class EditingForm

        Private Sub EditingForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Dim context As New Ch8DataContext(My.Settings.liaConnectionString)
            Dim query = From p In context.Publishers _
                                    Select p

            Me.BindingSource1.DataSource = query.ToList
        End Sub
    End Class
End Namespace